File: docs/examples/methodFunctions/on_event_halt.rs
use lightvm::LightVM;
use lightvm::types::{vmconfig::VmConfig, vmevent::VmEvent};

fn main() {
  let mut vm = LightVM::new(VmConfig {
    caps: vec![],
    ..Default::default()
  });

  vm.on(VmEvent::Halt, |data| {
    println!("Event: {:?}", data.event);
    println!("Payload: {:?}", data.payload);
  });
  vm.halt();
}

File: docs/examples/methodFunctions/onEventHalt.ts
import { LightVM, VMEvent } from 'lightvm';

const vm = new LightVM({ caps: [] });

vm.on(VMEvent.Halt, (data) => {
  console.log('Event: ', data.event);
  console.log('Payload: ', data.payload);
});
vm.halt();
vm.run(); // will not be executed
